From: Juhani Numminen <juhaninumminen0@gmail.com>
Date: Wed, 20 May 2015 21:15:10 +0300
Subject: kde-frameworks-5

The KDE thumbnailer plugin can now be compiled with KDE Frameworks 5

Origin: backport, http://sourceforge.net/p/pentobi/code/ci/8d783c0383746696010a186140e8b9cb91466732/
---
 CMakeLists.txt                                     |  5 +---
 src/libpentobi_kde_thumbnailer/CMakeLists.txt      |  8 +++++-
 src/pentobi_kde_thumbnailer/CMakeLists.txt         | 33 +++++++++++++++++++++-
 .../PentobiThumbCreator.cpp                        |  2 +-
 .../pentobi-thumbnail.desktop                      |  2 +-
 5 files changed, 42 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6cd0068..2b8c55a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.9)
+cmake_minimum_required(VERSION 2.8.12)
 
 # Use all lower-case for project name because it is used in
 # CMAKE_INSTALL_DOCDIR of GNUInstallDirs
@@ -24,9 +24,6 @@ option(PENTOBI_BUILD_QML "Build QtQuick-based GUI" OFF)
 option(PENTOBI_BUILD_KDE_THUMBNAILER "Build thumbnailer for KDE" OFF)
 option(USE_QT5 "Use Qt 5 instead of Qt 4" OFF)
 
-if (PENTOBI_BUILD_KDE_THUMBNAILER AND USE_QT5)
-  message(FATAL_ERROR "PENTOBI_BUILD_KDE_THUMBNAILER requires USE_QT5=0")
-endif()
 if (PENTOBI_BUILD_KDE_THUMBNAILER AND NOT PENTOBI_BUILD_GUI)
   message(FATAL_ERROR
     "PENTOBI_BUILD_KDE_THUMBNAILER requires PENTOBI_BUILD_GUI=1")
diff --git a/src/libpentobi_kde_thumbnailer/CMakeLists.txt b/src/libpentobi_kde_thumbnailer/CMakeLists.txt
index 6c877ea..222be9d 100644
--- a/src/libpentobi_kde_thumbnailer/CMakeLists.txt
+++ b/src/libpentobi_kde_thumbnailer/CMakeLists.txt
@@ -12,7 +12,9 @@
 # exceptions (which should be fine as long as no exceptions are thrown
 # from the thumbnailer plugin functions).
 
-include(${QT_USE_FILE})
+if (NOT USE_QT5)
+  include(${QT_USE_FILE})
+endif()
 
 add_definitions(${CMAKE_SHARED_LIBRARY_CXX_FLAGS})
 
@@ -40,3 +42,7 @@ set(pentobi_kde_thumbnailer_STAT_SRCS
 add_library(pentobi_kde_thumbnailer STATIC
   ${pentobi_kde_thumbnailer_STAT_SRCS}
 )
+
+if (USE_QT5)
+  target_link_libraries(pentobi_kde_thumbnailer Qt5::Widgets)
+endif()
diff --git a/src/pentobi_kde_thumbnailer/CMakeLists.txt b/src/pentobi_kde_thumbnailer/CMakeLists.txt
index cc371b5..2cc6791 100644
--- a/src/pentobi_kde_thumbnailer/CMakeLists.txt
+++ b/src/pentobi_kde_thumbnailer/CMakeLists.txt
@@ -1,3 +1,31 @@
+if (USE_QT5)
+
+
+find_package(ECM REQUIRED NO_MODULE)
+set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
+
+include(KDEInstallDirs)
+include(KDECompilerSettings)
+include(KDECMakeSettings)
+
+find_package(KF5 REQUIRED COMPONENTS KIO)
+
+include_directories(${CMAKE_SOURCE_DIR}/src)
+
+add_library(pentobi-thumbnail MODULE
+  PentobiThumbCreator.h
+  PentobiThumbCreator.cpp
+)
+
+target_link_libraries(pentobi-thumbnail PRIVATE
+  pentobi_kde_thumbnailer
+  KF5::KIOWidgets
+)
+
+
+else(USE_QT5)
+
+
 find_package(KDE4 REQUIRED)
 
 include(KDE4Defaults)
@@ -20,6 +48,9 @@ target_link_libraries(pentobi-thumbnail
   ${QT_QTCORE_LIBRARY}
   )
 
-install(TARGETS pentobi-thumbnail DESTINATION ${PLUGIN_INSTALL_DIR})
 
+endif(USE_QT5)
+
+
+install(TARGETS pentobi-thumbnail DESTINATION ${PLUGIN_INSTALL_DIR})
 install(FILES pentobi-thumbnail.desktop DESTINATION ${SERVICES_INSTALL_DIR})
diff --git a/src/pentobi_kde_thumbnailer/PentobiThumbCreator.cpp b/src/pentobi_kde_thumbnailer/PentobiThumbCreator.cpp
index 97a845d..d5695ed 100644
--- a/src/pentobi_kde_thumbnailer/PentobiThumbCreator.cpp
+++ b/src/pentobi_kde_thumbnailer/PentobiThumbCreator.cpp
@@ -15,7 +15,7 @@ using namespace std;
 
 extern "C"
 {
-    KDE_EXPORT ThumbCreator* new_creator()
+    Q_DECL_EXPORT ThumbCreator* new_creator()
     {
         return new PentobiThumbCreator;
     }
diff --git a/src/pentobi_kde_thumbnailer/pentobi-thumbnail.desktop b/src/pentobi_kde_thumbnailer/pentobi-thumbnail.desktop
index 576184f..1acd6dc 100644
--- a/src/pentobi_kde_thumbnailer/pentobi-thumbnail.desktop
+++ b/src/pentobi_kde_thumbnailer/pentobi-thumbnail.desktop
@@ -2,6 +2,6 @@
 Type=Service
 Name=Blokus games
 Name[de]=Blokus-Partien
-X-KDE-ServiceTypes=ThumbCreator
+ServiceTypes=ThumbCreator
 MimeType=application/x-blokus-sgf;
 X-KDE-Library=pentobi-thumbnail
